crates/catalog/rest/src/catalog.rs

```diff
@@ -745,10 +749,86 @@ impl Catalog for RestCatalog {
         _table_ident: &TableIdent,
         _metadata_location: String,
     ) -> Result<Table> {
-        Err(Error::new(
-            ErrorKind::FeatureUnsupported,
-            "Registering a table is not supported yet",
-        ))
+        let context = self.context().await?;
+
+        let request = context
+            .client
+            .request(
+                Method::POST,
+                context
+                    .config
+                    .register_table_endpoint(_table_ident.namespace()),
+            )
+            .json(&RegisterTableRequest {
+                name: _table_ident.name.clone(),
+                metadata_location: _metadata_location.clone(),
+            })
+            .build()?;
+
+        let http_response = context.client.query_catalog(request).await?;
+
+        let response: LoadTableResponse = match http_response.status() {
+            StatusCode::OK => {
+                deserialize_catalog_response::<LoadTableResponse>(http_response).await?
+            }
+            StatusCode::BAD_REQUEST => {
+                return Err(Error::new(
+                    ErrorKind::Unexpected,
+                    "Unexpected error while registering table.",
+                ));
+            }
+            StatusCode::UNAUTHORIZED => {
+                return Err(Error::new(
+                    ErrorKind::Unexpected,
+                    "Authenticated user does not have the necessary permissions.",
+                ));
+            }
+            StatusCode::FORBIDDEN => {
+                return Err(Error::new(
+                    ErrorKind::Unexpected,
+                    "Authenticated user does not have the necessary permissions.",
+                ));
+            }
+            StatusCode::NOT_FOUND => {
+                return Err(Error::new(
+                    ErrorKind::NamespaceNotFound,
+                    "The namespace specified does not exist.",
+                ));
+            }
+            StatusCode::CONFLICT => {
+                return Err(Error::new(
+                    ErrorKind::TableAlreadyExists,
```

Comment:
This should use `ErrorKind::CatalogCommitConflicts` and should be explicitly set to retryable using `with_retryable(true)`